rust/lance-encoding/src/encodings/logical/struct.rs

```diff
@@ -591,7 +571,21 @@ impl FieldEncoder for StructFieldEncoder {
         async move {
             let mut columns = Vec::new();
             // Add a column for the struct header
-            columns.push(EncodedColumn::default());
+            let mut header = EncodedColumn::default();
+            header.final_pages.push(EncodedPage {
+                array: EncodedArray {
+                    buffers: vec![],
+                    encoding: pb::ArrayEncoding {
+                        array_encoding: Some(pb::array_encoding::ArrayEncoding::Struct(
+                            pb::SimpleStruct {},
+                        )),
+                    },
+                },
+                num_rows: self.num_rows_seen,
+                column_idx: self.column_index,
+            });
```

Comment:
I'm a little surprised we are writing this unconditionally, given in most circumstances it wasn't necessary. Does it make sense to always be adding this?

Reply:
No.  It is, in fact, entirely unneeded at this point (I even considered fixing this by just removing the assert).  However, I'm in the middle of cleaning up struct/list encoding more thoroughly (aiming for 2.1) and want to minimize changes to the existing stuff if possible.

Reply:
(e.g. if I stop writing the column, that will break some of the reading code and I would need to make the reading code handle both cases then because there are enough 2.0 files in the wild I don't think we can afford breaking changes anymore)